### PR TITLE
Persist Whiteboard Pen Color 

### LINKED ISSUE #7834: [Feature Request] - Add setting option to choose "Default Whiteboard Pen Color"
Because of the background of my note types, I usually use the red pen color in the whiteboard. But every single time I start reviewing a deck, I have to choose the whiteboard pen color again, as the default blue has a bad contrast in my note types.

I think it would be very cool if a setting option was added to "Settings > Reviewing  > Whiteboard" section, so that another default pen color could be chosen. 

Another good solution would be the last chosen color be saved (& remembered) for the specific deck, just as enabling whiteboard currently is.

Thanks!

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
User requested

## Fixes
Fixes #7834 

## Approach
Persist the value in MetaDB, group by deck/light mode

## How Has This Been Tested?

Unit Tested (including a full run) - brief emulator test

## Learning 

* I'm glad we're better at testing, but it might have added too much friction to this change
* Adding new columns to MetaDB could be much easier - consider an alternate library for the far future.

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/WhiteboardDefaultForegroundColorTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    @ParameterizedRobolectricTestRunner.Parameter()
+    public boolean mIsInverted;
+
+    @ParameterizedRobolectricTestRunner.Parameter(1)
+    public boolean mIsMonochrome;
+
+    @ParameterizedRobolectricTestRunner.Parameter(2)
+    public int mExpectedResult;
+
+    @ParameterizedRobolectricTestRunner.Parameters
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/anki/WhiteboardDefaultForegroundColorTest.java:43]: look at you all fancy and parameterized ;-)

File: AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.java [hunk 2]
```diff
@@ -96,6 +96,9 @@ public void setUp() {
 
         //See: #6140 - This global ideally shouldn't exist, but it will cause crashes if set.
         DialogHandler.discardMessage();
+
+        // BUG: We do not reset the MetaDB
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.java:100]: interesting - what's the ideal resolution here? more specifically, what does "reset" mean? I thought robolectric used sqlite implementation from robolectric, configured for in-memory db and the tests were isolated so each test got it's own db clean?

**Review by mikehardy — APPROVED** [review]
this is a lot of heavy lifting in the test are yes, I feel like on any high coverage project I occasionally do a PR that is like 10x more test effort than code effort, but even on those PRs I so frequently catch a little thing at the end because of the testing it's worth it right then, and then over the life of the project for maintenance...oh I would never trade it. But it does take real time, for sure

I had one question in here but it's nothing that would stop approval or merge. Marked pending and I'll merge it in my next per-commit / full API sync merge fest (should happen shortly)

File: AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.java (hunk 2 above)
Comment (david-allison-1) [reply AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.java:100]: >  I thought robolectric used sqlite implementation from robolectric, configured for in-memory db and the tests were isolated so each test got it's own db clean?

The tests weren't isolated - this is possibly due to the static variable referencing the database in `MetaDB` (which we now reset - in the .closeDb call), but it warrants investigation.

File: AnkiDroid/src/test/java/com/ichi2/anki/WhiteboardDefaultForegroundColorTest.java (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/test/java/com/ichi2/anki/WhiteboardDefaultForegroundColorTest.java:43]: I need to find/write a framework for this. 

`ParameterizedRobolectricTestRunner` has a horrible API. Compare it to:

https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcase.html

Comment (mikehardy) [reply AnkiDroid/src/test/java/com/ichi2/anki/WhiteboardDefaultForegroundColorTest.java:43]: 🤔  at some point we're just going to konvert to the kool new way, it doesn't seem so bad in this case https://medium.com/@harmittaa/parametrized-tests-with-robolectric-3666c4794f6b

Comment (david-allison-1) [reply AnkiDroid/src/test/java/com/ichi2/anki/WhiteboardDefaultForegroundColorTest.java:43]: https://github.com/Pragmatists/JUnitParams

pls

Comment (mikehardy) [reply AnkiDroid/src/test/java/com/ichi2/anki/WhiteboardDefaultForegroundColorTest.java:43]: yeah no https://github.com/Pragmatists/JUnitParams/issues/169
oh, kotlin is koming, it's inevitable I think. I'm thinking of a konversion probably in the 2.17ish time frame, expecting 2.15/2.16 will be soaked up by rust infra

**Pull request merged**